app/models/agent.rb

```diff
@@ -129,12 +129,18 @@ def reload
   end
 
   def new_event_expiration_date
-    keep_events_for > 0 ? keep_events_for.days.from_now : nil
+    if keep_events_for == 60
```

Comment:
How about something like this?

```ruby
class Agent < ActiveRecord::Base
  EVENT_RETENTION_SCHEDULES = [["Forever", 0], ['1 hour', 1.hour], ['6 hours', 6.hours], ["1 day", 1.day], *([2, 3, 4, 5, 7, 14, 21, 30, 45, 90, 180, 365].map {|n| ["#{n} days", n.days] })]

  def new_event_expiration_date
    keep_events_for > 0 ? keep_events_for.seconds.from_now : nil
  end

  def update_event_expirations!
    if keep_events_for == 0
      events.update_all :expires_at => nil
    else
      events.update_all "expires_at = " + rdbms_date_add("created_at", "SECOND", keep_events_for.to_i)
    end
  end
end
```

And then also add a migration that updates all Agent's `keep_events_for` values:

`db/migrate/20150507153436_update_keep_events_for_to_be_in_seconds.rb`

```ruby
class UpdateKeepEventsForToBeInSeconds < ActiveRecord::Migration
  class Agent < ActiveRecord::Base; end

  SECONDS_IN_DAY = 60 * 60 * 24

  def up
    Agent.update_all ['keep_events_for = keep_events_for * ?', SECONDS_IN_DAY]
  end

  def down
    Agent.update_all ['keep_events_for = keep_events_for / ?', SECONDS_IN_DAY]
  end
end
```